modules/apps/27-interchain-accounts/host/keeper/msg_server.go

```diff
@@ -0,0 +1,37 @@
+package keeper
+
+import (
+	"context"
+
+	"cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/ibc-go/v7/modules/apps/27-interchain-accounts/host/types"
+	ibcerrors "github.com/cosmos/ibc-go/v7/modules/core/errors"
+)
+
+var _ types.MsgServer = (*msgServer)(nil)
+
+type msgServer struct {
+	*Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the ICS27 host MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper *Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
```

Comment:
I believe this construction was only used for the controller submodule to avoid name collision of the `SendTx` and `RegisterInterchainAccount` functions (one function signature is legacy/deprecated now)

Maybe we should just use the keeper directly?

Reply:
I'm kind of indifferent on this - using the unexported `msgServer` type does reduce the surface area for issues of other random modules doing something they shouldn't be 🤷